miden/tests/integration/operations/stack_ops.rs

```diff
@@ -28,3 +28,407 @@ fn swapdw() {
+        expected_values.append(&mut b);
+        expected_values.reverse();
+        build_op_test!(asm_op, &test_values).prop_expect_stack(&expected_values)?;
+    }
+
+    #[test]
+    fn movup_proptest(test_values in prop::collection::vec(any::<u64>(), 16), movup_idx in 2usize..16) {
+        let asm_op = format!("movup.{movup_idx}");
+        let mut expected_values = test_values.to_vec();
+        let movup_value = expected_values[15 - movup_idx];
+        expected_values.remove(15 - movup_idx);
+        expected_values.push(movup_value);
+        expected_values.reverse();
+        build_op_test!(asm_op, &test_values).prop_expect_stack(&expected_values)?;
+    }
+
+    #[test]
+    fn movupw_proptest(test_values in prop::collection::vec(any::<u64>(), 16), movupw_idx in 2usize..4) {
+        let asm_op = format!("movupw.{movupw_idx}");
+        let idx1 = 16 - (movupw_idx + 1) * 4;
+        let idx2 = 16 - movupw_idx * 4;
+        let mut movupw_values = test_values[idx1..idx2].to_vec();
+        let mut expected_values = test_values[..idx1].to_vec();
+        expected_values.append(&mut test_values[idx2..].to_vec());
+        expected_values.append(&mut movupw_values);
+        expected_values.reverse();
+        build_op_test!(asm_op, &test_values).prop_expect_stack(&expected_values)?;
+    }
+
+    #[test]
+    fn movdn_proptest(test_values in prop::collection::vec(any::<u64>(), 16), movdn_idx in 2usize..16) {
+        let asm_op = format!("movdn.{movdn_idx}");
+        let mut expected_values = test_values.to_vec();
+        let movdn_value = expected_values[15];
+        expected_values.remove(15);
+        expected_values.insert(15 - movdn_idx, movdn_value);
+        expected_values.reverse();
+        build_op_test!(asm_op, &test_values).prop_expect_stack(&expected_values)?;
+    }
+
+    #[test]
+    fn movdnw_proptest(test_values in prop::collection::vec(any::<u64>(), 16), movdnw_idx in 2usize..4) {
+        let asm_op = format!("movdnw.{movdnw_idx}");
+        let idx1 = 16 - (movdnw_idx + 1) * 4;
+        let mut movdnw_values = test_values[12..].to_vec();
+        let mut expected_values = test_values[..idx1].to_vec();
+        expected_values.append(&mut movdnw_values);
+        expected_values.append(&mut test_values[idx1..12].to_vec());
+        expected_values.reverse();
+        build_op_test!(asm_op, &test_values).prop_expect_stack(&expected_values)?;
+    }
+
+    #[test]
+    fn cswap_proptest(test_values in prop::collection::vec(any::<u64>(), 15), c in 0u64..2) {
+        let asm_op = "cswap";
+        let mut test_values_mut = test_values;
+        test_values_mut.push(c);
+        let mut expected_values = test_values_mut.to_vec();
+        expected_values.reverse();
+        if c == 1 {
+            expected_values.swap(1, 2)
+        };
+        expected_values.remove(0);
+        expected_values.push(0);
+        build_op_test!(asm_op, &test_values_mut).prop_expect_stack(&expected_values)?;
+    }
```

Comment:
You can declare `mut test_values` in the signature instead of using the extra variable test_values_mut, e.g.:

```
    fn cswap_proptest(mut test_values in prop::collection::vec(any::<u64>(), 15), c in 0_u64..2) {
        let asm_op = "cswap";
        test_values.push(c);
        let mut expected_values = test_values.clone();
```